src/reducers/index.ts

```diff
@@ -52,21 +52,51 @@ function checkStoreInvariants(state: StoreState) {
     state.currentMatchIndex >= -1 &&
       state.currentMatchIndex <= state.matchesList.length
   );
+
+  checkCondition(
+    'every matchId in matchIdToMatchState is also present in matchesList',
+    Object.keys(state.matchIdToMatchState).reduce(
+      (a, e) =>
```

Comment:
Better to give them readable names then a and e (even though they prev and curr)